Pinta.Core/Classes/ToolOption/IntegerOption.cs

```diff
@@ -0,0 +1,131 @@
+//
+// IntegerOption.cs
+//
+// Author:
+//       Paul Korecky <https://github.com/spaghetti22>
+//
+// Copyright (c) 2025 Paul Korecky
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+
+namespace Pinta.Core;
+
+/// <summary>
+/// An integer-valued option that creates a spin button with a set minimum and maximum.
+/// </summary>
+public class IntegerOption : ToolOption
+{
+	private string name;
+	public int Minimum { get; private set; }
+	public int Maximum { get; private set; }
+	public string LabelText { get; private set; }
+
+	// this is intentionally an "int" and not a "long" because the settings manager
+	// only supports System.Int32 - if a wider type is needed, also extend the 
+	// settings manager so that settings can be saved there
+	private int value;
+	private List<ValueChange> value_change_callbacks;
```

Comment:
Could this be replaced by an `event`? It seems to kind of be replicating the same idea of registering multiple callbacks